src/Datagrid/Pager.php

```diff
@@ -68,8 +63,13 @@ public function init(): void
     {
         $this->resultsCount = $this->computeResultsCount();
 
-        $this->getQuery()->setFirstResult(null);
-        $this->getQuery()->setMaxResults(null);
+        $query = $this->getQuery();
+        if (null === $query) {
```

Comment:
Probably out of scope of this Pr, but wouldnt be nice if we ensure somehow that the query is never null, maybe as a parameter of this method or as a parameter of construct?

Reply:
Currently the pager and the query are constructor param of the Datagrid.
In this one we're running
```
$this->pager->setQuery($this->query);
$this->pager->init();
```

Maybe we could remove setQuery and use the query as a param of init indeed.